### PR TITLE
Fixes Pinned Inspector and containers.

### LINKED ISSUE #7455: It is possible to modify a Prefab entity without being in its Focus Mode using the Pinned Inspector
**Describe the bug**
If the User opens the _Pinned Inspector_ for an entity inside the Prefab when being in its _Focus Mode_, the entity can still be edited even after exiting the _Focus Mode_.

Please note that editing the entity this way does not dirty the Prefab (there is no indication of unsaved changes on the Prefab). Also, there is no _Save Prefab to file_ option in the right-click context menu of the Prefab.

Please refer to the attached video for more details.

**Steps to reproduce**
1. Open any Level.
2. Create an entity.
3. Make a Prefab out of the entity.
4. Enter the _Focus Mode_ of the Prefab.
5. RMB the entity inside the Prefab and select _Open Pinned Inspector_.
6. Exit the _Focus Mode_ of the Prefab.
7. Add a new Component to the entity using the _Pinned Inspector_

**Expected behavior**
It is not possible to modify an entity inside the Prefab without being in the Prefab's _Focus Mode_.

**Actual behavior**
It is possible to modify an entity inside the Prefab without being in the Prefab's _Focus Mode_ using the _Pinned Inspector_.

**Assets required**
N/A

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/152804699-24d748ce-8d31-4b75-abb1-c735f03225f3.mp4

**Found in Branch**
Development (b1df4c9)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 SUPER
 - Memory 16GB

### PR DESCRIPTION
**Pull request by jjjoness**
Signed-off-by: John Jones-Steele <82226755+jjjoness@users.noreply.github.com>
Fixes #7455

### CODE REVIEW

**Review by AMZN-daimini — COMMENTED** [review]
Code looks good, but can you add a screenshot of the Inspector when it gets disabled? Just want to understand if the Inspector is still visible but greyed out/disabled, or it just turns completely blank.

**Comment by jjjoness** [comment]
image
@AMZN-daimini here is a screen shot.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp [hunk 1]
```diff
@@ -5666,15 +5672,40 @@ namespace AzToolsFramework
...
         AzToolsFramework::PropertyEditorEntityChangeNotificationBus::MultiHandler::BusDisconnect(entityId);
     }
 
+    void EntityPropertyEditor::OnEditorFocusChanged(
+        [[maybe_unused]] AZ::EntityId previousFocusEntityId, [[maybe_unused]] AZ::EntityId newFocusEntityId)
+    {
+        // We need to wait for the Container Entity System to update before checking
+        QTimer::singleShot( 1, this, [this]()
+            {
+                bool visible = false;
```
[the remark this answers is not in the document]
Comment (jjjoness) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp:5695]: Changed.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp [hunk 2]
```diff
@@ -5666,15 +5672,40 @@ namespace AzToolsFramework
...
+                for (AZ::EntityId entityId : m_overrideSelectedEntityIds)
+                {
+                    if (m_focusModeInterface->IsInFocusSubTree(entityId) &&
+                        !m_containerEntityInterface->IsUnderClosedContainerEntity(entityId))
+                    {
+                        visible = true;
+                        break;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (jjjoness) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp:5704]: Yes, better this way. Changed.

**Review by AMZN-daimini — APPROVED** [review]
Looks good to me!

Only additional input I have is that there should be more messaging about the difference between a pinned inspector and the main one, and maybe a message on why it is disabled? But I would consider those changes for a follow-up, this is enough to address the issue at hand.

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**